src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -2131,7 +2131,7 @@ private void ReplaceAllInChunk(int[] replacements, int replacementsCount, String
                     for (;;)
                     {
                         // Copy in the new string for the ith replacement
-                        ReplaceInPlaceAtChunk(ref targetChunk, ref targetIndexInChunk, valuePtr, value.Length);
+                        ReplaceInPlaceAtChunk(ref targetChunk!, ref targetIndexInChunk, valuePtr, value.Length);
```

Comment:
bangs in ReplaceInPlaceAtChunk shouldn't be needed anymore since you changed it to nullable, right?

Reply:
The thing here is that `targetChunk` is not nullable, but the parameter is nullable. Since this is a ref parameter, it is saying it could have a null value. That is why the `!` is needed.